File: back-end/Fretefy.Test.Domain/Dto/Regiao/RegiaoResponseDto.cs
using Fretefy.Test.Domain.Dto;
using Fretefy.Test.Domain.Entities;
using System;
using System.Collections.Generic;
using System.Text;

namespace Fretefy.Test.Application.Dto
{
    public class RegiaoResponseDto
    {
        public Guid Id { get; set; }
        public string Nome { get; set; }
        public List<Cidade> Cidades { get; set; }
    }
}

File: front-end/src/app/models/dto/cidade/regiao/regiao-response-dto.ts
import { Cidade } from 'src/app/models/cidade';
import { CidadeResponseDto } from '../cidade/cidade-response-dto'; 

export class RegiaoResponseDto {
  id: string;
  nome: string;
  cidades: Cidade[];

  constructor(id: string, nome: string, cidades: CidadeResponseDto[]) {
    this.id = id;
    this.nome = nome;
    this.cidades = cidades;
  }
}